File: hopital-backend/src/main/java/com/gestionhopital/hopital_backend/service/PatientService.java
package com.gestionhopital.hopital_backend.service;

import java.util.List;

import com.gestionhopital.hopital_backend.model.Patient;

public interface PatientService {
    Patient savePatient(Patient patient);
    Patient updatePatient(Long id, Patient patientDetails);
    void deletePatient(Long id);
    List<Patient> getAllPatients();
    Patient getPatientById(Long id);
    List<Patient> searchPatients(String keyword);
}

File: hopital-frontend/src/app/services/patient.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Patient } from '../models/patient.model';

@Injectable({
  providedIn: 'root'
})
export class PatientService {

  private apiUrl = '/api/patients'; 

  constructor(private http: HttpClient) { }

  getPatients(): Observable<Patient[]> {
    return this.http.get<Patient[]>(this.apiUrl);
  }

  getPatientById(id: number): Observable<Patient> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.get<Patient>(url);
  }

  createPatient(patient: Patient): Observable<Patient> {
    return this.http.post<Patient>(this.apiUrl, patient);
  }

  updatePatient(id: number, patientDetails: Patient): Observable<Patient> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.put<Patient>(url, patientDetails); 
  }


  deletePatient(id: number): Observable<any> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.delete(url); 
  }
}